integration/common/openlineage/common/provider/dbt.py

```diff
@@ -0,0 +1,202 @@
+            return run_results
+
+    @staticmethod
+    def load_yaml(path: str) -> Dict:
+        with open(path, 'r') as f:
+            return yaml.load(f)
+
+    def parse_artifacts(self, manifest: Dict, run_results: Dict) -> List[DbtRun]:
+        nodes = {}
+        runs = []
+
+        # Filter non-model nodes
+        for name, node in manifest['nodes'].items():
+            if name.startswith('model.'):
+                nodes[name] = node
+
+        for run in run_results['results']:
+
+            def get_timings(timings: List[Dict]) -> Tuple[str, str]:
+                timing = list(filter(lambda x: x['name'] == 'execute', timings))[0]
+                return timing['started_at'], timing['completed_at']
+
+            started_at, completed_at = get_timings(run['timing'])
+            inputs = [
+                nodes[node]
+                for node
+                in manifest['parent_map'][run['unique_id']]
+                if node.startswith('model.')
+            ] + [
+                manifest['sources'][source]
+                for source
+                in manifest['parent_map'][run['unique_id']]
+                if source.startswith('source.')
+            ]
+
+            runs.append(DbtRun(
+                started_at,
+                completed_at,
+                run['status'],
+                inputs,
+                nodes[run['unique_id']],
+                run['unique_id'],
+                self.namespace
+            ))
+        return runs
+
+    def to_openlineage_events(self, run: DbtRun) -> Tuple[RunEvent, RunEvent]:
+        try:
+            return RunEvent(
+                eventType=RunState.START,
+                eventTime=run.started_at,
+                run=Run(
+                    runId=run.run_id
+                ),
+                job=Job(
+                    namespace=self.namespace,
+                    name=run.job_name
+                ),
+                producer=PRODUCER,
+                inputs=[self.node_to_dataset(node) for node in run.inputs],
+                outputs=[self.node_to_dataset(run.output)]
+            ), RunEvent(
+                eventType=RunState.COMPLETE,
+                eventTime=run.completed_at,
+                run=Run(
+                    runId=run.run_id
+                ),
+                job=Job(
+                    namespace=self.namespace,
+                    name=run.job_name,
+                    facets={
+                        'sql': SqlJobFacet(run.output['compiled_sql'])
+                    }
+                ),
+                producer=PRODUCER,
+                inputs=[self.node_to_dataset(node, facets=True) for node in run.inputs],
+                outputs=[self.node_to_dataset(run.output, facets=True)]
+            )
+        except Exception as e:
+            raise ValueError(e)
+
+    def node_to_dataset(self, node: Dict, facets: bool = False) -> Dataset:
+        if facets:
```

Comment:
I would rename the boolean `facets` to `has_facets` (or similar) to avoid confusing the variable with dict `facets`.

Reply:
Done.